build/dsc/resource.rb

```diff
@@ -82,7 +82,26 @@ def ensurable?
 
     def absentable?
       properties.detect do |p|
-        p.is_ensure? && p.values.any? { |v| v.casecmp('absent') == 0 }
+        return 'absent' if p.is_ensure? && p.values.any? { |v| v.casecmp('absent') }
```

Comment:
Just realized that the logic here is not quite right as well... this changes the previous `true` / `false` return values to something different.

Will fix in my updates.